contrib/intel/jenkins/tests.py

```diff
@@ -688,58 +693,100 @@ def execute_cmd(self):
+            os.chdir(self.pwd)
 
     @property
     def execute_condn(self):
-        return (self.mpi_type == 'impi' or \
-               (self.mpi_type == 'mpich' and self.core_prov == 'verbs'))
-
-    def execute_cmd(self, testgroupname):
-        print("Running Tests: " + testgroupname)
-        tests = []
-        time = None
-        os.chdir(f'{self.mpichsuitepath}/{testgroupname}')
-        tests = self.testgroup(testgroupname)
-        for test in tests:
-            testname = test[0]
-            nprocs = test[1]
-            args = test[2:]
-            for item in args:
-               itemlist =  item.split('=')
-               if (itemlist[0] == 'timelimit'):
-                   time = itemlist[1]
-            self.set_options(nprocs, timeout=time)
-            testcmd = f'./{testname}'
-            outputcmd = shlex.split(self.mpi.env + self.mpi.cmd + testcmd + '\'')
-            common.run_command(outputcmd)
-        os.chdir(self.pwd)
-
+        return ((self.mpi_type == 'impi' or \
+                self.mpi_type == 'mpich') and \
+               (self.core_prov == 'verbs' or self.core_prov == 'tcp'))
+    def execute_cmd(self):
+        if (self.mpi_type == 'mpich'):
+            configure_cmd = './configure'
+            if (self.weekly):
+                print(f'Weekly {self.mpi_type} mpichsuite tests')
+                os.chdir(self.mpichsuitepath)
+                common.run_command(shlex.split(self.mpi.env + configure_cmd +'\''))
+                self.exclude_tests(self.mpichsuitepath, self.core_prov)
+                testcmd = 'make testing'
+                outputcmd = shlex.split(self.mpi.env +testcmd + '\'')
+                common.run_command(outputcmd)
+                os.chdir(self.pwd)
+            else:
+                print(f"PR {self.mpi_type} mpichsuite tests")
+                os.chdir(self.mpichsuitepath)
+                configure_cmd += f' --with-mpi={cloudbees_config.build_dir}/mpich_temp'
+                common.run_command(shlex.split(configure_cmd))
+                common.run_command(['make', '-j'])
+                self.exclude_tests(self.mpichsuitepath, self.core_prov)
+                testcmd = "./runtests -tests=testlist -debug -verbose"
+                common.run_command(shlex.split(self.mpi.env +testcmd + '\''))
+                os.chdir(self.pwd)
+        if (self.mpi_type == 'impi' and self.weekly == True):
+            print (f'Weekly {self.mpi_type} mpichsuite tests')
+            os.chdir(self.impi_mpichtestpath)
+            print(self.hosts)
+            self.create_hostfile(f'{self.impi_mpichtestpath}/hostfile',
+                                    self.hosts)
+            os.environ["I_MPI_HYDRA_HOST_FILE"] = \
+                                    f'{self.impi_mpichtestpath}/hostfile'
+            test_root = f'{self.impi_mpichtestpath}/test/mpi'
+            prov = self.core_prov
+            test_cmd = f'export I_MPI_HYDRA_HOST_FILE={self.impi_mpichtestpath}/hostfile; '
```

Comment:
this can be abstracted into a make_test_cmd function where the options can be split to be added/removed more easily in the future

Reply:
the make_test_cmd will have to again switch over which type of mpi and type of job(PR/weekly) in order to create a unique command for each case